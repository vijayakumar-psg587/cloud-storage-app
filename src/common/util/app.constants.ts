export const APP_NAME = 'CLOUD_AUTHENTICATOR';
export const APP_DEFAULT_TIMESTAMP_FORMAT = 'YYYY-DD-MM hh:mm:ss.SSSS';
export const APP_SERVER_ZONE  = 'America/New_York';
export const CHAR_HYPEN = '-';
export const CHAR_ZERO = '0';
export const CHAR_COLON = ':';
export const APP_MESSAGE_KEY = 'cloud-storage-message-key';
export const APP_LOG_REDACT = [];
export const APP_LOG_CENSOR = '**REDACTED';
export const GCLOUD_KEY_FILE = '/home/vijaykumar/Documents/glcoud/service-accounts/dev-index-254608-03_17_2020.json';
export const GCLOUD_PRJ_ID = 'dev-index-254608';
export const GCLOUD_STROAGE_SCOPE = ['https://www.googleapis.com/auth/devstorage.read_write','https://www.googleapis.com/auth/pubsub'];
export const GCLOUD_CONN_MAX_RETRIES = 3;
export const GCLOUD_ENV_STR_BUCKET_NAME = 'feedback-database-connector';
export const GCLOUD_UPLOAD_FILE_DEV_LOCAL_PATH = '/home/vijaykumar/Documents/Code/Nestjs/database-connector/config/development/dev.txt';
export const GCLOUD_UPLOAD_FILE_PROD_LOCAL_PATH = '/home/vijaykumar/Documents/Code/Nestjs/database-connector/config/production/prod.txt';
export const GCLOUD_DATABASE_BUCKET_DEV = 'mongo-data/development/dev.txt';
export const GCLOUD_DATABASE_BUCKET_PROD = 'mongo-data/production/prod.txt';
export const ENV_NAME_DEV = 'dev';
export const GCLOUD_STORED_FILE_NAME_DEV = 'dev.txt';
export const GCLOUD_STORED_FILE_NAME_PROD = 'prod.txt';

