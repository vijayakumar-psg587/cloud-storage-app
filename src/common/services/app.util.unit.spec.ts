import { AppUtilServiceInstance } from './app.util.service';
import momentTime from 'moment-timezone';
import mock from "mock-fs";

jest.mock('moment-timezone', () =>
	jest.fn().mockImplementation( ()=> (
		{
			format: () => '2018–01–30 12:34:56.0000'
		})));
const globalAny: any = global;
describe('Test suite for util service', () => {
	beforeEach(() => {
		jest.restoreAllMocks();
	});
	// afterAll(() => {
	// 	jest.clearAllMocks();
	// 	jest.resetModules();
	// 	mock.restore();
	// });
	test('testing getCurrentLocalTime', () => {
		const localTime: string = AppUtilServiceInstance.getCurrentLocaleTimeZone();
		const utcString = new Date(Date.now() * 1000).toUTCString();

		expect(localTime).toBeDefined();
		expect(localTime.substr(localTime.lastIndexOf(':') + 1, localTime.length - 1)).toHaveLength(2);
		expect(localTime.substr(0, localTime.indexOf('-'))).toHaveLength(5);
		expect(localTime.substr(localTime.lastIndexOf('-') + 1, 2)).toHaveLength(2);
	});
	test('testing general getUTC time', () => {

		const timestamp = AppUtilServiceInstance.getGeneralUTCTimeStamp();
		expect(timestamp).toMatchSnapshot();
	});

	test('testing isDevEnv', () => {
		process.env.NODE_ENV = 'dev';
		let flag = AppUtilServiceInstance.isDevEnv();
		expect(flag).toBeTruthy();
		process.env.NODE_ENV = globalAny.ENV_JEST;
		flag = AppUtilServiceInstance.isDevEnv();
		expect(flag).toBeFalsy();
	});

	test('testing isNullOrUndefined', () => {
		expect(AppUtilServiceInstance.isNullOrUndefined(null)).toBeTruthy();
		expect(AppUtilServiceInstance.isNullOrUndefined(Object.create({}))).toBeFalsy();
	});
});

