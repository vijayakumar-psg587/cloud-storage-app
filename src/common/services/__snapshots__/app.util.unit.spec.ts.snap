// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for util service testing general getUTC time 1`] = `"2018–01–30 12:34:56.0000"`;
